clients/python-wrapper/lakefs/branch.py

```diff
@@ -417,11 +437,15 @@ def __exit__(self, typ, value, traceback) -> bool:
 
         try:
             self._tx_branch.commit(message=self._tx.commit_message, metadata=self._tx.commit_metadata)
-            self._tx_branch.merge_into(self._source_branch, message=f"Merge transaction {self._tx.id} to branch")
+            merge_commit_id = self._tx_branch.merge_into(
+                self._source_branch, message=f"Merge transaction {self._tx.id} to branch")
             self._tx_branch.delete()
-
-            return False
         except LakeFSException as e:
             if self._cleanup_branch:
                 self._tx_branch.delete()
             raise TransactionException(f"Failed committing transaction {self._tx.id}: {e}") from e
+
+        if self._tx.tag:
+            tag = Tag(self._repo_id, self._tx.tag, self._client)
+            tag.create(merge_commit_id)
```

Comment:
A few open question here:

1. What should happen if the tag already exists? Should a new tag be force-created?
2. What should happen if the tag creation fails? Does the transaction needs to be reverted? I don't think we should but just asking.

Reply:
To clarify - tag creation failure should not require revert of transaction and commit and merge should be successful as well as tx_branch cleanup if applicable. We should somehow notify the user on the failure

Reply:
I think the location you added the code is good